### DISCUSSION TITLE
Strapi password reset link not working & email validation

### DISCUSSION THREAD

**Discussion by zizi-busy** [strapi/strapi#7381] [Legacy Topics]
I cannot access any of my content any longer :/ I am a total newbie so here is something for you to laugh while I cry:

I ran strapi locally for my project , had to reinstall it. However, I was asked to set up a new account (sign in option didn't work over and over again!)
But here comes the fun part,  when I finally managed to get to the login screen I couldn't remember which email I used!! #facepalm (i rely on my pc saving this stuff but somehow with strapi it didn't happen). Not a biggie, I thought, I will just check strapi confirmation email. But there was none.I thought it is not a big deal, strapi email validation will let me know if the email is already in use or I can reset password and such.. I tried one of my emails but ended up setting up a new account, so that couldn't be the one I already used (or could it ?! hmm) .. with another possible email I tried the password reset but never got the password reset link.. so here I am, wanting to cry, cuz i spent SO MUCH TIME on this, I have so much content in strapi yet I  ended up with empty strapi account x( I am losing hope that anyone can help me with this mess up as password reset doesn't work.. :( Even though this is all my fault, I believe Stapi should improve form validation! I tried typing 2897h@gmail.com and password reset was sent to non -existing email.   Makes me think maybe I typed in my email wrongly.. No confirmation link..How did I miss that... Rant over

**Comment by jeremy21212121** [comment]
You need to configure a third-party email provider in order to send email to your gmail address. If you didn't do that, then that is why you aren't receiving emails.

Check out the [config section of the email plugin documentation](https://strapi.io/documentation/v3.x/plugins/email.html#configure-the-plugin). Here is the relevant quote:

> By default Strapi provides a local email system ([sendmail](https://www.npmjs.com/package/sendmail)). If you want to use a third party to send emails, you need to install the correct provider module.

Let's unpack this a little bit. By default, strapi uses a _local_ email system. This is not email as most people know it. This means that emails can't be sent outside of your own computer. With linux, email can be sent between users on the same system. If you want to send emails to your gmail address, you will need to set up a third party service, such as sendgrid, as described in the [docs here](https://strapi.io/documentation/v3.x/plugins/email.html#configure-your-provider). Sendgrid, for example, has a free plan, so it shouldn't cost you anything.

Now onto recovering the email address you used. By default, strapi uses sqlite as database. Sqlite uses a regular file to store data. The strapi default DB location is `.tmp/data.db`. That means that the email address you used is stored in that file, we just need to know how to get it out. The easiest way of doing this is probably to use the `sqlite3` CLI tool. Hopefully it is installed already but if not, the way to install it will depend on which OS you are using.

Open a terminal (possibly powershell or WSL on windows) and change directory into your strapi project.

Run `sqlite3 .tmp/data.db` and you should see a sqlite prompt. Now we can run commands on the database to get the info we need.

At the sqlite prompt, type `.tables` and enter. This will list all the tables in the database and should look something like this:

```
sqlite> .tables
core_store                    strapi_webhooks             
posts                         upload_file                 
strapi_administrator          upload_file_morph           
strapi_permission             users-permissions_permission
strapi_role                   users-permissions_role      
strapi_users_roles            users-permissions_user
```

The table we are interested in is "strapi-administrator". Next we will run a SQL query to get the contents of the email fields.

Run `SELECT email from "strapi_administrator";`. That will return the contents of the email field for each row in that table.

I hope that helps! Best of luck, please let me know how that works for you.

**Reply by zizi-busy** [reply]
Thank you so much for this!

**Reply by lauriejim** [reply]
Thank you for this full answer, I really appreciate your help 💪
